docs/community/maintainers-guide/releasing.rst

```diff
@@ -0,0 +1,224 @@
+
+1. Check that all relevant PRs have been merged. For the ``invenio`` package,
+   check that all relevant Invenio modules have been released.
+2. Fetch latest changes:
+    - ``git fetch inveniosoftware``
+3. Checkout maintenance or master branch:
+    - ``git checkout maint-<major>.<minor>``
+4. Merge or reset to latest changes:
+    - ``git merge --ff-only inveniosoftware/maint-<major>.<minor>``
+    - or
+    - ``git reset --hard inveniosoftware/maint-<major>.<minor>``
+
+While above is not strictly necessary, it avoids situations where you local
+and GitHub branches have diverged.
+
+Patch-level release
+-------------------
+Most of the time, patch-level releases are not necessary for the main
+``invenio`` package, since by patch-level releases of the underlying
+modules are automatically picked up: the dependencies upper bounds are
+normally allowing the installation of patch-level releases.
+
+Most of the Invenio modules have maintenance branches for previous
+major/minor releases. They are normally named ``maint-<version>``.
+
+With patch-level releases, you'll often make multiple releases and work with
+multiple maintenance branches. For instance, you might have to merge a
+bug fix to ``master``, but also to ``maint-3.1`` and to ``maint-3.2``.
+
+Major/minor-level release
+-------------------------
+When releasing a major/minor version, you will have to take extra care to the
+compatibility with the current released ``invenio`` version, by checking
+the upper bounds defined for this module in the ``invenio`` releases.
+
+The most probable scenario is that for a given ``invenio`` release,
+e.g. ``3.3``, the upper bound for an Invenio module ``foo``
+(with current latest release ``1.2.6``) is set to ``invenio-foo<1.3.0``.
+A new minor release ``1.3.0`` of this module will not affect
+``invenio 3.3``.
+
+If this is not the case, you will have to make sure that the new release
+of the module ``foo`` will not introduce backward incompatible changes,
+because it will be installed by ``invenio 3.3``, potentially breaking
+existing Invenio installations.
+
+Module release
+--------------
+
+How to release an Invenio module.
+
+**Python module**
+
+1. Make sure that all relevant PRs have been merged to ``master``.
+2. Make sure that the build on ``master`` branch is passing.
+3. Prepare release notes in
+    - Update the ``CHANGE.rst`` file, making sure that you describe changes,
+      highlighting backward incompatible ones, and that the release date
+      is correct. Pay attention how you write, keep consistency with previous
+      sentences forms.
+4. Bump version in ``version.py`` (follow semantic versioning).
+5. Create a release commit (message pattern:
+   ``release: v<major>.<minor>.<patch>``).
+6. Create tag and push release:
+    - ``git tag v<major>.<minor>.<patch>``
+    - ``git push inveniosoftware master v<major>.<minor>.<patch>``
+7. Regularly check ``Travis`` and ``PyPI`` to ensure the release went through.
```

Comment:
What about saying something along the lines of:
```
Confirm that the Travis build run successfully and that the PyPI release is completed, e.g. https://travis-ci.org/github/inveniosoftware/invenio-records-rest/builds/684646338 and https://pypi.org/project/invenio-records-rest/1.6.5/
```

Reply:
personally I think we are adding too many links and it might be difficult to follow (e.g. get lost opening tabs). That said, I dont have a strong opinion against that, so it can be added. WDYT?

Reply:
Kind of agree on this one, maintainer should know how to find Travis or PyPI pages?